packages/core/src/tweening/spring.ts

```diff
@@ -0,0 +1,116 @@
+import {decorate, threadable} from '../decorators';
+import {ThreadGenerator} from '../threading';
+import {useProject, useThread, useLogger} from '../utils';
+
+decorate(spring, threadable());
+export function* spring(
+  spring: Spring | null,
+  from: number,
+  to: number,
+  settleTolerance: number | null,
+  onProgress: (value: number, time: number) => void,
+  onEnd?: (value: number, time: number) => void,
+): ThreadGenerator {
+  spring = spring ?? {
+    mass: 0.05,
+    stiffness: 10,
+    damping: 0.5,
+  };
+
+  if (spring.mass <= 0) {
+    useLogger().error(new Error('Spring mass must be greater 0.'));
+    return;
+  }
+  if (spring.stiffness < 0) {
+    useLogger().error(
+      new Error('Spring stiffness must be greater or equal to 0.'),
+    );
+    return;
+  }
+  if (spring.damping < 0) {
+    useLogger().error(
+      new Error('Spring damping must be greater or equal to 0.'),
+    );
+    return;
+  }
+
+  const project = useProject();
+  const thread = useThread();
+
+  const startTime = thread.time();
+
+  // figure out the step length for each frame
+  const timeStep = 1 / project.framerate;
+
+  settleTolerance = settleTolerance ?? 0.001;
+  let position = from;
+  let velocity = spring.initalVelocity ?? 0;
+  let settled = false;
+
+  onProgress(from, 0);
+  while (!settled) {
+    const positionDelta = position - to;
+
+    // using hooks law: F=-kx; with k being the spring constant and x the offset to the settling position
+    const force = -spring.stiffness * positionDelta - spring.damping * velocity;
+
+    //updating the velocity
+    velocity += (force / spring.mass) * timeStep;
+
+    // apply velocity to the position
+    position += velocity * timeStep;
+
+    // the simulation would run forever but is often barely noticable after some seconds
+    // we fix this by checking if the delta delta between $to and $position is smaller than the given $settleTolerance
+    // but we also need to check if the velocity is below the tolerance too
```

Comment:
Unfortunately, prettier won't do this for you, but comments should also be limited to 80 characters.
Also, a few minor mistakes in spelling/grammar:
```suggestion
    // The simulation would run forever but is often barely noticeable after
    // some seconds. We fix this by checking if the delta between `to` and
    // `position` is smaller than the given `settleTolerance`. But we also need
    // to check if the velocity is below the tolerance too.
```